content/gsoc/2024/_index.md

```diff
@@ -32,30 +37,64 @@ We provide a [template for proposals](https://docs.google.com/document/d/1gER-yF
 
 ## Projects
 
-### Project title 1
+### Migrate from `unittest` to `pytest`
+
+PyBaMM's inception predates the rise in popularity of `pytest`, and so we have used the `unittest` testing framework for our tests. However, `pytest` is now the de facto standard for testing in Python, and it is time to migrate our test cases over to it. The reason for this is that `pytest` is much more flexible and powerful than `unittest`, and will allow us to write better tests with less code. This may also involve migrating our test cases to use `hypothesis` for property-based testing, which will allow us to write even more powerful tests via the use of strategies, which are like generators for test cases.
+
+PyBaMM already uses `pytest` with the `nbmake` package to run our example notebooks as tests, but still uses `unittest` for the rest of the tests. `pytest-xdist` is also used to run the tests in parallel, which should end up being faster than the current serial implementation and should be adapted to be used with the new test framework.
 
-Project description
+The student will be expected to look at other popular Python packages to see how they use `pytest` and `hypothesis` to write tests, and then migrate our tests over to use these frameworks. This will involve writing new tests, and also migrating existing tests over to the new framework, making use of and exploring the `pytest` documentation for scouting potential features to be incorporated. The student will also be expected to improve the test coverage of the codebase, and to write tests for any new code that they end up writing.
+
+As a stretch goal, the packaging infrastructure for PyBaMM should be updated to include a `pytest` test suite that can be run by users to check that their installation is working correctly for users who do not have access to a source installation of PyBaMM, i.e., through `pybamm.test()` or similar.
 
 #### Expected outcomes
 
-* Expected outcome 1
+* All test files and modules migrated to use `pytest` in the code base, in a way that is consistent with modern Python software engineering and testing practices – either manually or via automated migration tools
+* Improvisation of test coverage
+* Utilisation of advanced testing techniques based on advanced `pytest` features and `hypothesis` as a framework
+* Deprecate some legacy test invocation methods such as `run-tests.py` in favour of the `pytest` path finding and test case discovery mechanisms
 
 #### Desired skills
 
-* Desired skill 1
-* Desired skill 2
-* Desired skill 3
+* Experience with Python API design and testing, test-driven development, and continuous integration
+* Test coverage and code quality analysis tools
+* Some experience with `unittest`, `pytest`, and `hypothesis` is desirable, but not required
+* Git version control, managing pull requests, and code reviews, as well as the GitHub workflow for open-source projects
```

Comment:
Remember, GSoC is meant for Open-Source beginners, haha!
```suggestion
* Experience with Python and unit testing (`unittest`, `pytest`, and `hypothesis`) in Python.
* Some experience with test-driven development, continuous integration, and coverage tools is desirable, but not required.
* Git version control and GitHub workflow for open-source projects.
```

Reply:
Makes sense!